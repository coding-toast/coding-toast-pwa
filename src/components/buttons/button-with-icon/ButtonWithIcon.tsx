import React from 'react'
import { Link } from 'gatsby'
import ButtonWithIconStyles from './ButtonWithIcon.module.scss'

interface ButtonWithIconProps {
  toUrl: string
}

export const ButtonWithIcon: React.FC<ButtonWithIconProps> = props => {
  const { toUrl } = props
  return (
    <Link to={toUrl} className={`${ButtonWithIconStyles.button} text-style-label`}>
      Read More
    </Link>
  )
}
