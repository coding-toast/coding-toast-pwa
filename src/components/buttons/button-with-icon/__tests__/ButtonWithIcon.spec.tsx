import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ButtonWithIcon } from '../ButtonWithIcon'

describe('<ButtonWithIcon />', () => {
  it('renders correctly without children', () => {
    const component = shallow(<ButtonWithIcon toUrl="/" />)
    expect(toJson(component)).toMatchSnapshot()
  })
})
