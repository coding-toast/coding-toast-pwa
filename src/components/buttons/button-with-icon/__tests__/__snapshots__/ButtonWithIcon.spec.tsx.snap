// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonWithIcon /> renders correctly without children 1`] = `
<mockConstructor
  className="button text-style-label"
  to="/"
>
  Read More
</mockConstructor>
`;
